parquet-common/src/main/java/org/apache/parquet/io/SeekableInputStream.java

```diff
@@ -105,4 +107,21 @@ public abstract class SeekableInputStream extends InputStream {
    */
   public abstract void readFully(ByteBuffer buf) throws IOException;
 
+  /**
+   * Read a set of file ranges in a vectored manner.
```

Comment:
Can we get some param docs here?  The behavior of the `allocate` param wasn't immediately clear to me.

Reply:
+1

Reply:
this has been covered with validation of overlaps and invalid ranges; hadoop is consistent with this too.